framework/src/executioners/Transient.C

```diff
@@ -257,31 +257,35 @@ Transient::init()
   }
 
   _problem.initialSetup();
+
   _time_stepper->init();
 
   if (_app.isRestarting())
     _time_old = _time;
 
   _problem.outputStep(EXEC_INITIAL);
 
-  // If this is the first step
-  if (_t_step == 0)
-    _t_step = 1;
+  if (_app.isRecovering()) // Recover case
+  {
+    if (_t_step == 0)
+      mooseError("cannot at time step zero when recovering");
 
-  if (_t_step > 1) // Recover case
     _dt_old = _dt;
-
+  }
   else
   {
+    if (_t_step != 0)
+      mooseError("time step must be zero during initialization without recovering");
+
     computeDT();
-    //  _dt = computeConstrainedDT();
     _dt = getDT();
-  }
-  if (_dt == 0)
-    mooseError("Initial time step size is not evaluated properly");
+    if (_dt == 0)
+      mooseError("Initial time step size is not evaluated properly");
```

Comment:
Again, nobody will have any clue what went wrong and why.

Reply:
This is probably cause by a stepper bug. I can rephrase if desired.